src/werkzeug/datastructures.py

```diff
@@ -1279,11 +1292,22 @@ def update(self, *args, **kwargs):
             raise TypeError("update expected at most 1 arguments, got %d" % len(args))
 
         if args:
-            for key, value in iter_multi_items(args[0]):
-                self[key] = value
+            mapping = args[0]
+        else:
+            mapping = kwargs
```

Comment:
`dict.update()` allows an object and kwargs at the same time. This change makes it either or.

Reply:
Thanks, fixed.